src/store/mod.rs

```diff
@@ -38,6 +38,22 @@ mod writer;
 pub use self::reader::StoreReader;
 pub use self::writer::StoreWriter;
 
+trait Compression {
+    fn compress(uncompressed: &[u8], compress: &mut Vec<u8>);
+    fn decompress(compressed: &[u8], decompressed: &mut Vec<u8>);
+}
+
+#[cfg(feature="lz4")]
+mod compression_lz4;
+#[cfg(feature="lz4")]
+use self::compression_lz4::*;
+
+#[cfg(not(feature="lz4"))]
+mod compression_snap;
+#[cfg(not(feature="lz4"))]
+use self::compression_snap::*;
```

Comment:
Cool, so by default we get [snappy](http://google.github.io/snappy/), but someone can turn on lz4 with the feature flag. Our snappy lib is a pure rust implementation, but `lz4` requires a C library. Sweet.

Reply:
yes... LZ4 is probably a nicer compresson/performance trade-off but not by a large margin.
This makes it easier to compile tantivy to web assembly.